### PR TITLE
Write unit tests for the organization management feature and implement the organization switching feature

### PR DESCRIPTION
**Pull request by thivi**
1. Organization List
2. Add ORganziation Modal
3. Organisation Page

In addition, this PR also implements the organization switching logic.

### Related Issues
- Issue `#1` or (None)

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/organizations/__tests__/organization-list.test.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                <AccessControlProvider
+                    allowedScopes={ ReduxStoreStateMock.auth.scope }
+                    featureConfig={ ReduxStoreStateMock.config.ui }
+                >
+                    <OrganizationList { ...organizationListProps } />
+                </AccessControlProvider>
+            </Provider>
+        );
+
+        expect(screen.getByText("Organization One"));
```
Comment (thivi) [inline apps/console/src/features/organizations/__tests__/organization-list.test.tsx:118]: Get the number of organizations shown in the list and validate it.

File: apps/console/src/features/organizations/__tests__/organization-list.test.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    <OrganizationList { ...organizationListProps } />
+                </AccessControlProvider>
+            </Provider>
+        );
+
+        fireEvent.click(screen.getAllByTestId("data-table-row")[ 0 ]);
+        expect(onListItemClickMock.mock.calls.length).toBe(1);
+    });
+
+    test("UTC-1.7 - Test if an organizations can be deleted", async () => {
```
Comment (thivi) [inline apps/console/src/features/organizations/__tests__/organization-list.test.tsx:137]:
```suggestion
    test("UTC-1.7 - Test if an organization can be deleted", async () => {
```

File: apps/console/src/features/organizations/__tests__/organization.test.tsx [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+        await waitFor(() => expect(getOrganizationsMock).toHaveBeenCalledTimes(1));
+
+        expect(await screen.findByText("Organization One")).toBeInTheDocument();
+        expect(await screen.findByText("Organization Two")).toBeInTheDocument();
+        expect(await screen.findByText("Organization Three")).toBeInTheDocument();
+        expect(await screen.findByText("Organization Four")).toBeInTheDocument();
+        expect(await screen.findByText("Organization Five")).toBeInTheDocument();
+        expect(await screen.findByText("Organization Six")).toBeInTheDocument();
+        expect(await screen.findByText("Organization Seven")).toBeInTheDocument();
```
Comment (thivi) [inline apps/console/src/features/organizations/__tests__/organization.test.tsx:88]: Validate the number of organizations.

File: apps/console/src/features/organizations/__tests__/organization.test.tsx [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+        expect(
+            await within(await screen.findByTestId("organizations-breadcrumb")).findByText("Organization One")
+        ).toBeInTheDocument();
+        expect(
+            await within(await screen.findByTestId("organizations-breadcrumb")).findByTestId(
+                "organizations-breadcrumb-home"
+            )
+        ).toBeInTheDocument();
+        expect(await screen.findByText("Organization Two")).toBeInTheDocument();
```
Comment (thivi) [inline apps/console/src/features/organizations/__tests__/organization.test.tsx:113]: Validate the number of organizations.

File: apps/console/src/features/organizations/__tests__/organization.test.tsx [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+        ).toBeInTheDocument();
+        expect(
+            await within(await screen.findByTestId("organizations-breadcrumb")).findByText("Organization Two")
+        ).toBeInTheDocument();
+        expect(
+            await within(await screen.findByTestId("organizations-breadcrumb")).findByTestId(
+                "organizations-breadcrumb-home"
+            )
+        ).toBeInTheDocument();
+        expect(await screen.findByText("Organization Three")).toBeInTheDocument();
```
Comment (thivi) [inline apps/console/src/features/organizations/__tests__/organization.test.tsx:142]: Validate the number of organizations.

File: apps/console/src/features/organizations/__tests__/organization.test.tsx [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+        expect(
+            await within(await screen.findByTestId("organizations-breadcrumb")).findByText("Organization One")
+        ).toBeInTheDocument();
+        expect(
+            await within(await screen.findByTestId("organizations-breadcrumb")).findByTestId(
+                "organizations-breadcrumb-home"
+            )
+        ).toBeInTheDocument();
+        expect(await screen.findByText("Organization Two")).toBeInTheDocument();
```
Comment (thivi) [inline apps/console/src/features/organizations/__tests__/organization.test.tsx:172]: Validate the number of organizations.

File: apps/console/src/features/organizations/components/organization-switch/organization-switch-dropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,347 @@
+/**
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com). All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
```
Comment (thivi) [inline apps/console/src/features/organizations/components/organization-switch/organization-switch-dropdown.tsx:8]: Update the license header

File: apps/console/src/features/organizations/components/organization-switch/organization-switch-dropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,347 @@
+/**
```
Comment (thivi) [inline apps/console/src/features/organizations/components/organization-switch/organization-switch-dropdown.tsx:1]: Add data component ids

File: apps/console/src/features/organizations/components/organization-switch/organization-switch-dropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+                            className="name ellipsis tenant-description"
+                            data-testid={ "organization-dropdown-display-name" }
+                        >
+                            { organization?.name ?? (
+                                <Placeholder>
+                                    <Placeholder.Line />
+                                </Placeholder>
+                            ) }
+
+                            { organization.id !== "ROOT" && (
```
Comment (thivi) [inline apps/console/src/features/organizations/components/organization-switch/organization-switch-dropdown.tsx:165]: Root organization id has been updated

File: apps/console/src/features/organizations/constants/organization-constants.ts [hunk 10]
```diff
@@ -29,6 +31,18 @@ export class OrganizationManagementConstants {
         .set("ORGANIZATION_DELETE", "organizations.delete")
         .set("ORGANIZATION_READ", "organizations.read");
 
+    /**
+     * Root organization object.
+     *
+     * @@constant
```
Comment (thivi) [inline apps/console/src/features/organizations/constants/organization-constants.ts:37]:
```suggestion
     * @constant
```

File: apps/console/src/app.tsx [hunk 11]
```diff
@@ -92,6 +94,24 @@ export const App: FunctionComponent<Record<string, never>> = (): ReactElement =>
         sessionStorageDisabled();
     }, []);
 
+    useEffect(() => {
+        if (!window[ "AppUtils" ].getConfig().organizationName || !config.endpoints.organizations) {
```
Comment (brionmario) [inline apps/console/src/app.tsx:98]: Shall we get this from redux?

File: apps/console/src/app.tsx [hunk 12]
```diff
@@ -92,6 +94,24 @@ export const App: FunctionComponent<Record<string, never>> = (): ReactElement =>
...
+            dispatch(setGetOrganizationLoading(false));
+
+            return;
+        }
+
+        const orgName = window[ "AppUtils" ].getConfig().organizationName;
+
+        dispatch(setGetOrganizationLoading(true));
+        getOrganizations(`id eq ${ orgName }`, 1, null, null, true).then((response: OrganizationListInterface) => {
```
[the remark this answers is not in the document]
Comment (thivi) [reply apps/console/src/app.tsx:109]: Resolved by https://github.com/wso2/identity-apps/pull/3136/commits/67d2c09348f434982fb710846034a419918517a6

**Pull request merged**